plugins/httpapi/aci.py

```diff
@@ -0,0 +1,191 @@
+        self.check_auth_from_private_key = None
+        self.backup_hosts = None
+        self.host_counter = 0
+
+    def set_params(self, auth, params):
+        self.params = params
+        self.auth = auth
+
+    def set_backup_hosts(self):
+        try:
+            list_of_hosts = re.sub(r'[[\]]', '', self.connection.get_option("host")).split(",")
+            ipaddress.ip_address(list_of_hosts[0])
+            return list_of_hosts
+        except Exception:
+            return []
+
+    def login(self, username, password):
+        ''' Log in to APIC '''
+        # Perform login request
+        self.connection.queue_message('log', 'Establishing connection to {0}'.format(self.connection.get_option('host')))
+        method = 'POST'
+        path = '/api/aaaLogin.json'
+        payload = {'aaaUser': {'attributes': {'name': username, 'pwd': password}}}
+        data = json.dumps(payload)
+        try:
+            response, response_data = self.connection.send(path, data, method=method)
+            response_value = self._get_response_value(response_data)
+            self.connection._auth = {'Cookie': 'APIC-Cookie={0}'
+                                        .format(self._response_to_json(response_value).get('imdata')[0]['aaaLogin']['attributes']['token'])}
+            self.connection.queue_message('vvvv', 'Establishing connection to {0} was successful'.format(self.connection.get_option('host')))
+        except Exception:
+            self.connection.queue_message('vvvv', 'Failed establishing connection to {0}'.format(self.connection.get_option('host')))
+            self.handle_error()
+
+    def logout(self):
+        method = 'POST'
+        path = '/api/aaaLogout.json'
+        try:
+            response, response_data = self.connection.send(path, {}, method=method)
+        except Exception as e:
+            msg = 'Error on attempt to logout from APIC. {0}'.format(e)
+            raise ConnectionError(self._return_info(None, method, path, msg))
+        self.connection._auth = None
+        self._verify_response(response, method, path, response_data)
+
+    def send_request(self, method, path, json):
+        ''' This method handles all APIC REST API requests other than login '''
+        if json is None:
+            json = {}
+        # Case1: List of hosts is provided
+        # if self.host_counter == 0:
+        #     with open('/tmp/hosts.pkl', 'wb') as fi:
+        #         pickle.dump(0, fi)
+        self.backup_hosts = self.set_backup_hosts()
+        #self.connection.set_option('persistent_command_timeout', 3)
+        if not self.backup_hosts:
+            # if self.connection._connected is True and self.params.get('host') != self.connection.get_option("host"):
+            #     self.connection._connected = False
+            if self.params.get('host') is not None:
+                self.connection.set_option("host", self.params.get('host'))
+
+            if self.params.get('port') is not None:
```

Comment:
send_request() function is not complete yet because I need more APIC sims.
The general idea is to utilize the ip from aci_host if ansible_host is not provided in our inventory and set parameters based off of aci_host. As of now this is not required. I only added these statements for testing purposes.